tests/ops/test_stats.py

```diff
@@ -1,12 +1,90 @@
 import pytest
 import torch
 
-from pyro.ops.stats import (autocorrelation, autocovariance, effective_sample_size,
-                            gelman_rubin, split_gelman_rubin,
-                            _cummin, _fft_next_good_size)
+from pyro.ops.stats import (autocorrelation, autocovariance, effective_sample_size, gelman_rubin,
+                            hpdi, pi, quantile, resample, split_gelman_rubin, waic, _cummin,
+                            _fft_next_good_size)
 from tests.common import assert_equal, xfail_if_not_implemented
 
 
+@pytest.mark.parametrize('replacement', [True, False])
+@pytest.mark.init(rng_seed=3)
+def test_resample(replacement):
+    x = torch.empty(10000, 2)
+    x[:, 0].normal_(3, 4)
+    x[:, 1].normal_(5, 6)
+
+    num_samples = 5000
+    y = resample(x, num_samples=num_samples, replacement=replacement)
+    z = resample(x.t(), num_samples=num_samples, dim=1, replacement=replacement)
+    if not replacement:
+        assert_equal(torch.unique(y.reshape(-1)).numel(), y.numel())
+        assert_equal(torch.unique(z.reshape(-1)).numel(), z.numel())
+    assert_equal(y.shape, torch.Size([num_samples, 2]))
+    assert_equal(z.shape, torch.Size([2, num_samples]))
+    assert_equal(y.mean(dim=0), torch.tensor([3., 5.]), prec=0.1)
+    assert_equal(z.mean(dim=1), torch.tensor([3., 5.]), prec=0.12)
+    assert_equal(y.std(dim=0), torch.tensor([4., 6.]), prec=0.1)
+    assert_equal(z.std(dim=1), torch.tensor([4., 6.]), prec=0.1)
+
+
+@pytest.mark.init(rng_seed=3)
```

Comment:
Why does this need any rng seed?

Reply:
@neerajprad `y` is generated uniformly. But I observed that the generated samples are flaky somehow (even if we use a number much larger than `2000`).